src/Integration.Vsix/SonarLintTagger/Scheduler.cs

```diff
@@ -38,29 +39,26 @@ public Scheduler()
             jobs = new Dictionary<string, WeakReference<CancellationTokenSource>>(StringComparer.OrdinalIgnoreCase);
         }
 
-        public void Schedule(string jobId, Action<CancellationToken> action, int? timeoutInMilliseconds)
+        public void Schedule(string jobId, Action<CancellationToken> action, int timeoutInMilliseconds)
         {
-            var newTokenSource = IssueToken(jobId);
+            var newCancellationToken = IssueToken(jobId, timeoutInMilliseconds);
 
-            if (timeoutInMilliseconds.HasValue)
-            {
-                newTokenSource.CancelAfter(timeoutInMilliseconds.Value);
-            }
-
-            action(newTokenSource.Token);
+            action(newCancellationToken);
```

Comment:
There was a race condition here all along, which got discovered due to CancelAfter: 
1. Thread A gets to IssueToken, acquires a lock and issues a new CancellationTokenSource.
2. Thread B gets to IssueToken, is blocked by thread A.
3. Thread A releases the lock, and proceeds to CancelAfter.
4. Thread B acquires a lock and issues a new CancellationTokenSource, cancelling the token issued by thread A.
5. Thread A proceeds to `action(newTokenSource.Token)` and receives an Object disposed exception.